test/apps/iodemo/run_io_demo.sh

```diff
@@ -414,238 +417,238 @@ make_scripts()
 
 		# Add file header and startup
 		cat >${command_file} <<-EOF
-			#!/bin/sh ${sh_args}
+			#!/bin/sh
 			#
 			# Launch script for io_demo on ${host} with ${num_servers_per_host[${host}]} servers and ${num_clients_per_host[${host}]} clients
 			#
 
-			kill_iodemo() {
-			    # kill process by exact name to avoid killing launch scirpts
+			list_pids() {
 			    # lt- prefix is in case iodemo is wrapped by libtool
 			    for pattern in ${exe_basename} lt-${exe_basename}
 			    do
-			        pkill --exact -u ${USER} -9 \${pattern}
+			        pgrep --exact -u ${USER} \${pattern}
 			    done
 			}
 
+			list_pids_with_role() {
+			    # list all process ids with role \$1
+			    for pid in \$(list_pids)
+			    do
+			        grep -qP "IODEMO_ROLE=\$1\x00" /proc/\${pid}/environ \\
+			            && echo \${pid}
+			    done
+			}
+
+			kill_iodemo() {
+			    pids="\$(list_pids)"
+			    [ -n "\${pids}" ] && kill -9 \${pids}
```

Comment:
btw, is there signal that iodemo_exe will respond with graceful exit?

Reply:
there is no such signal, we could add it and use kill -9 after some timeout (to ensure cleanup)